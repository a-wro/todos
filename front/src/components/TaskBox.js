import React from 'react'
import TaskForm from './forms/taskForm'
import Todo from './Todo'
import DeleteDialog from './DeleteDialog'
import { connect } from 'react-redux'
import { getTodos } from '../actions/getTodos'
import { postTodo } from '../actions/postTodo'
import { deleteTodo } from '../actions/deleteTodo'
import { editTask } from '../actions/updateTask'
import Paper from '@material-ui/core/Paper'
import Button from '@material-ui/core/Button'
import AddIcon from '@material-ui/icons/Add'
import CircularProgress from '@material-ui/core/CircularProgress'
import './styles/TaskBox.css'

export class TaskBox extends React.Component {

    componentDidMount() {
        this.props.getTodos()
    }

    state = {
        term: '',
        id: '', // active task in form
        dialogVisible: false,
        deleteDialogVisible: false,
        dialogType: null, // 'Add', 'Edit', 'Delete'
    }

    handleChange = e => {
        this.setState({ term: e.target.value })
    }

    handleSubmit = (e) => {
        e.preventDefault()
        if (this.state.dialogType === 'Add') {
            this.props.postTodo(this.state.term)
        }
        else {
            this.props.editTask(this.state.id, this.state.term)
        }

        this.setState({ dialogVisible: false })
    }

    openTaskForm = () => {
        this.setState({ dialogVisible: true, dialogType: 'Add', term: '' })
    }

    openEditForm = (task, id) => {
        this.setState({ dialogVisible: true, dialogType: 'Edit', term: task, id: id })
    }

    openDeleteDialog = (id) => {
        this.setState({ deleteDialogVisible: true, id: id})
    }

    closeTaskForm = () => {
        this.setState({ dialogVisible: false })
    }

    closeDeleteDialog = () => {
        this.setState({ deleteDialogVisible: false })
    }

    confirmDelete = () => {
        console.log(this.state.id)
        this.props.deleteTodo(this.state.id)
        this.setState({ deleteDialogVisible: false })
    }

    render() {
        const { todos, deleteTodo } = this.props
        if (todos.loading) {
            return (
                <div className="center">
                   <CircularProgress size={100}/>
                </div>
            )
        }

        return (
            <div className="task-box">
                <TaskForm
                    visible={this.state.dialogVisible}
                    type={this.state.dialogType}
                    handleClose={this.closeTaskForm}
                    onChange={this.handleChange}
                    value={this.state.term}
                    onSubmit={this.handleSubmit}
                />

                <DeleteDialog
                    visible={this.state.deleteDialogVisible}
                    handleClose={this.closeDeleteDialog}
                    confirmDelete={this.confirmDelete}
                />


                  {todos.tasks.map(task => {
                      return (
                        <Paper key={task.id} className="todo-paper">
                            <Todo
                                className="task"
                                id={task.id}
                                task={task.task}
                                completed={task.completed}
                                openEditForm={this.openEditForm}
                                openDeleteDialog={this.openDeleteDialog}
                            />
                        </Paper>
                      )
                  })}

                <div className="add-button">
                    <Button
                        onClick={this.openTaskForm}
                        variant="fab"
                        color="primary"
                        aria-label="Add">
                        <AddIcon />
                    </Button>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        todos: state.todos

    }
}

export default connect(mapStateToProps, { getTodos, postTodo, editTask, deleteTodo })(TaskBox)