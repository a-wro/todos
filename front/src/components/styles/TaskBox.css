
.add-button {
    text-align: center;
    padding-top: 1.25rem;
}

.todo-paper {
    margin: auto;
    width: 60%;
    padding-top: 1.25rem;
    padding-bottom: 1.25rem;
    height: 100%;
    margin-top: 0.5rem;
}

.task-box {
    margin-top: 2.5rem;
}