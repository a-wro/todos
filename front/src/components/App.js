import React, { Component } from 'react'
import TaskBox from './TaskBox'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from '../reducers/'
import Grid from  '@material-ui/core/Grid'
import ChecklistBox from './ChecklistBox';

const store = createStore(rootReducer, applyMiddleware(thunk))

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <Grid container spacing={24}>
            <Grid item xs={12}>
            <ChecklistBox />
            
            </Grid>
            <Grid item xs={6} ys={2}>
            <div>

            </div>
            </Grid>
          </Grid>
        </div>
      </Provider>
    );
  }
}

export default App;
