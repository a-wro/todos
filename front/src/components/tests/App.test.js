import React from 'react';
import App from '../App';
import TaskBox from '../TaskBox'
import { shallow } from 'enzyme'

describe('App', () => {
  const app = shallow(<App />)
  describe('Rendering', () => {
    it('renders without crashing', () => {
      expect(app).toMatchSnapshot()
    })
    
    it('correctly renders a `TaskBox` component', () => {
      expect(app.find(TaskBox).exists()).toBe(true)
      
    })
  
  
  
  })
})
