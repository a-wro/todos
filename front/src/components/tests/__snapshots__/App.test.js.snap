// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
>
        <WithStyles(Grid)
                container={true}
                spacing={24}
        >
                <WithStyles(Grid)
                        item={true}
                        xs={12}
                >
                        <Connect(ChecklistBox) />
                </WithStyles(Grid)>
                <WithStyles(Grid)
                        item={true}
                        xs={6}
                        ys={2}
                >
                        <div />
                </WithStyles(Grid)>
        </WithStyles(Grid)>
</div>,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Grid)
          container={true}
          spacing={24}
>
          <WithStyles(Grid)
                    item={true}
                    xs={12}
          >
                    <Connect(ChecklistBox) />
          </WithStyles(Grid)>
          <WithStyles(Grid)
                    item={true}
                    xs={6}
                    ys={2}
          >
                    <div />
          </WithStyles(Grid)>
</WithStyles(Grid)>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Grid)
              item={true}
              xs={12}
>
              <Connect(ChecklistBox) />
</WithStyles(Grid)>,
            <WithStyles(Grid)
              item={true}
              xs={6}
              ys={2}
>
              <div />
</WithStyles(Grid)>,
          ],
          "container": true,
          "spacing": 24,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(ChecklistBox) />,
              "item": true,
              "xs": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <div />,
              "item": true,
              "xs": 6,
              "ys": 2,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
>
          <WithStyles(Grid)
                    container={true}
                    spacing={24}
          >
                    <WithStyles(Grid)
                              item={true}
                              xs={12}
                    >
                              <Connect(ChecklistBox) />
                    </WithStyles(Grid)>
                    <WithStyles(Grid)
                              item={true}
                              xs={6}
                              ys={2}
                    >
                              <div />
                    </WithStyles(Grid)>
          </WithStyles(Grid)>
</div>,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <WithStyles(Grid)
            container={true}
            spacing={24}
>
            <WithStyles(Grid)
                        item={true}
                        xs={12}
            >
                        <Connect(ChecklistBox) />
            </WithStyles(Grid)>
            <WithStyles(Grid)
                        item={true}
                        xs={6}
                        ys={2}
            >
                        <div />
            </WithStyles(Grid)>
</WithStyles(Grid)>,
          "className": "App",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Grid)
                item={true}
                xs={12}
>
                <Connect(ChecklistBox) />
</WithStyles(Grid)>,
              <WithStyles(Grid)
                item={true}
                xs={6}
                ys={2}
>
                <div />
</WithStyles(Grid)>,
            ],
            "container": true,
            "spacing": 24,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Connect(ChecklistBox) />,
                "item": true,
                "xs": 12,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div />,
                "item": true,
                "xs": 6,
                "ys": 2,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
